services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/stage2/assign_students.tsx

```diff
@@ -0,0 +1,356 @@
+          </div>
+          <div className="import-or-create-classroom-buttons">
+            {renderImportFromProviderButton(canvasProvider)}
+            {renderImportFromProviderButton(cleverProvider)}
+            {renderImportFromProviderButton(googleProvider)}
+            <button
+              className="quill-button medium secondary outlined create-a-class-button"
+              onClick={() => openModal(createAClassForm)}
+              type="button"
+            >
+            Create a class
+            </button>
+          </div>
+        </div>
+        <div className="assignment-section-body">
+          {renderCreateAClassInlineForm()}
+          {renderAllClassroomsCheckbox()}
+          {renderClassroomList()}
+        </div>
+      </div>
+    )
+  }
+
+  const renderCreateAClassInlineForm = () => {
+    return visibleModal === createAClassForm
+      ? <CreateAClassInlineForm cancel={closeModal} onSuccess={onSuccess} />
+      : null
+  }
+
+  const renderImportFromProviderButton = (theProvider: string) => {
+    if (provider && provider != theProvider) { return null }
+
+    const theProviderTitle = providerLookup[theProvider].title
+
+    let buttonContent = <React.Fragment>Import from {theProviderTitle}</React.Fragment>
+    let buttonClassName = "interactive-wrapper import-from-provider-button"
+
+    if (providerClassroomsLoading && pendingImportFromProviderRequest) {
+      buttonContent = <React.Fragment>Import from {theProviderTitle}<ButtonLoadingIndicator /></React.Fragment>
+      buttonClassName += ' loading'
+    }
+
+    const importFromTheProvider = importFromProvider[theProvider]
+    const alt = `${theProviderTitle} Icon`
+    const src = providerIconSrc[theProvider]
+
+    return (
+      <button className={buttonClassName} onClick={importFromTheProvider} type="button">
+        <img alt={alt} className='import-from-provider-button-icon' src={src} />
+        {buttonContent}
+      </button>
+    )
+  }
+
+  const renderImportProviderClassroomsModal = () => {
+    if (visibleModal !== importProviderClassroomsModal) { return null }
+
+    return (
+      <ImportProviderClassroomsModal
+        classrooms={providerClassrooms}
+        close={() => closeModal(() => setPendingImportFromProviderRequest(false))}
+        onSuccess={onSuccess}
+        provider={provider}
+        user={user}
+      />
+    )
+  }
+
+  const renderLinkProviderAccountModal = () => {
+    let link = ''
+    let linkAccountProvider = ''
+
+    if (visibleModal === linkCanvasAccountModal) {
+      link = canvasLink
+      linkAccountProvider = canvasProvider
+    } else if (visibleModal === linkCleverAccountModal) {
+      link = cleverLink
+      linkAccountProvider = cleverProvider
+    } else if (visibleModal === linkGoogleAccountModal) {
+      // no link assignment since google uses a different component for linking accounts
+      linkAccountProvider = googleProvider
+    } else {
+      return null
+    }
```

Comment:
nbd but I'd probably use a `switch` statement here since you're evaluating the same variable every time.